R/informedbinomialtestbayesian.R

```diff
@@ -0,0 +1,257 @@
+  jaspResults[["models"]] <- models
+
+  if (length(options[["models"]]) > 0)
+    startProgressbar(length(options[["models"]]))
+
+
+  modelsList <- list()
+
+  # fit an overall unrestricted model (for plotting the posterior)
+  modelsList[[1]] <- list(
+    "model" = try(multibridge::binom_bf_informed(
+      x             = dataset[,options[["successes"]]],
+      n             = dataset[,options[["sampleSize"]]],
+      Hr            = paste0(levels(dataset[,options[["factor"]]]), collapse = ","),
+      a             = options[["priorCounts"]][[1]][["values"]],
+      b             = options[["priorCounts"]][[2]][["values"]],
+      factor_levels = dataset[,options[["factor"]]],
+      bf_type       = "BF0r",
+      nburnin       = options[["mcmcBurnin"]],
+      niter         = options[["mcmcBurnin"]] + options[["mcmcSamples"]],
+      maxiter       = options[["bridgeSamples"]],
+      seed          = if (options[["setSeed"]]) .getSeedJASP(options) else sample.int(.Machine$integer.max, 1),
+    )),
+    "name"  = "unrestricted"
+  )
+
+  # estimate the restricted models
+  for(i in seq_along(options[["models"]])) {
+
+    if (nchar(options[["models"]][[i]][["syntax"]]) == 0) {
+
+      modelsList[[i+1]] <- list(
+        "model" = NULL,
+        "name"  = options[["models"]][[i]][["modelName"]]
+      )
+
+    } else {
+
+      modelsList[[i+1]] <- list(
+        "model" = try(multibridge::binom_bf_informed(
+          x             = dataset[,options[["successes"]]],
+          n             = dataset[,options[["sampleSize"]]],
+          Hr            = options[["models"]][[i]][["syntax"]],
+          a             = options[["priorCounts"]][[1]][["values"]],
+          b             = options[["priorCounts"]][[2]][["values"]],
+          factor_levels = dataset[,options[["factor"]]],
+          bf_type       = "BF0r",
+          nburnin       = options[["mcmcBurnin"]],
+          niter         = options[["mcmcBurnin"]] + options[["mcmcSamples"]],
+          maxiter       = options[["bridgeSamples"]],
+          seed          = if (options[["setSeed"]]) .getSeedJASP(options) else sample.int(.Machine$integer.max, 1),
+        )),
+        "name"  = options[["models"]][[i]][["modelName"]]
+      )
+    }
+
+    progressbarTick()
+  }
+
+  models$object <- modelsList
+
+  return()
+}
+.createInformedBinBayesPosteriorPlot      <- function(jaspResults, dataset, options) {
+
+  if (!is.null(jaspResults[["posteriorPlot"]]))
+    return()
+
+  # extract posterior summary from the unrestricted model and format it for the plotting function
+  tempSummary           <- data.frame(
+    fact     = levels(dataset[,options[["factor"]]]),
+    observed = (options[["priorCounts"]][[1]][["values"]] + dataset[,options[["successes"]]]) / (options[["priorCounts"]][[1]][["values"]] + dataset[,options[["successes"]]] + options[["priorCounts"]][[2]][["values"]] + (dataset[,options[["sampleSize"]]]-dataset[,options[["successes"]]])),
```

Comment:
can you use `dataset[[...]]` everywhere you extract a single column from a dataframe?